src/main/java/com/flansmod/common/guns/EntityDamageSourceFlan.java

```diff
@@ -0,0 +1,78 @@
+package com.flansmod.common.guns;
+
+import com.flansmod.common.PlayerHandler;
+import com.flansmod.common.types.InfoType;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.EntityDamageSourceIndirect;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentString;
+
+public class EntityDamageSourceFlan extends EntityDamageSourceIndirect{
+	
+	private InfoType weapon;
+	private EntityPlayer shooter;
+	private boolean headshot;
+	/**
+	 * @param s        Name of the damage source (Usually the shortName of the gun)
+	 * @param entity   The Entity causing the damage (e.g. Grenade). Can be the same as 'player'
+	 * @param player   The Player responsible for the damage
+	 * @param wep      The InfoType of weapon used
+	 */	
+	public EntityDamageSourceFlan(String s, Entity entity, EntityPlayer player, InfoType wep)
+	{
+		this(s, entity, player, wep, false);
+	}
+	
+	/**
+	 * @param s        Name of the damage source (Usually the shortName of the gun)
+	 * @param entity   The Entity causing the damage (e.g. Grenade). Can be the same as 'player'
+	 * @param player   The Player responsible for the damage
+	 * @param wep      The InfoType of weapon used
+	 * @param headshot True if this was a headshot, false if not
+	 */
+	public EntityDamageSourceFlan(String s, Entity entity, EntityPlayer player, InfoType wep, boolean headshot)
+	{
+		super(s, entity, player);
+		weapon = wep;
+		shooter = player;
+		this.headshot = headshot;
+	}
+	
+	@Override
+	public ITextComponent getDeathMessage(EntityLivingBase living)
+	{
+		if(!(living instanceof EntityPlayer) || shooter == null || PlayerHandler.getPlayerData(shooter) == null)
+		{
+			return super.getDeathMessage(living);
+		}
+
+		return new TextComponentString("#flansmod");
```

Comment:
Prints "#flansmod" to server console whenever there's a flan's mod death, this should be replaced with an appropriate death message.

Reply:
Complicated. This is currently used to hide the DeathMessage when a FlansMod-Death occurs
(In 'ClientEventHandler.chatMessage(ClientChatReceivedEvent event)'). While it should be possible to make a vanilla style chat message with some effort, this would mean that this message is also displayed at the client side along the custom flansmod death messages.
This raises the problem of the lack of server side ForgeEvents to stop the server from sending (flansmod) chat death messages to the client, meaning a coremod would be required (or another workaround like the current one, but i would not favor such implementation)

Reply:
> > I was wondering if we could perhaps have a headshot icon that demonstrates "headshot" a bit better. Perhaps something like the image below where there's a Steve face as the layer below a scope reticle icon.
> 
> I am no pixel-art-designer, so i can not really provide any feedback on this.
> The current image was created by me, as i have not found anything suiting in the current FlansMod assets. The idea is, to have something matching the gun/grenade icons (16*16 pixels), but this drastically limits the amount of detail that can be included in the image.
> But as said, i will happily accept another matching solution for the image.

Fair enough! Perhaps someone will submit a new one in the future.

Reply:
I'll merge this PR and then we can look into the death message in dedicated servers issue.